src/status_im/ui/components/qr_code/views.cljs

```diff
@@ -0,0 +1,63 @@
+(ns status-im.ui.components.qr-code.views
+  (:require [re-frame.core :as re-frame]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.icons.vector-icons :as vi]
+            [status-im.ui.components.status-bar.view :as status-bar]
+            [status-im.ui.components.list-selection :as list-selection]
+            [status-im.ui.components.qr-code.styles :as styles]
+            [status-im.utils.platform :as platform]
+            [status-im.i18n :as i18n]
+            [status-im.react-native.js-dependencies :as rn-dependencies]
+            [reagent.core :as r])
+  (:require-macros [status-im.utils.views :refer [defview letsubs]]))
+
+(defn qr-code [props]
+  (r/create-element
+    rn-dependencies/qr-code
+    (clj->js (merge {:inverted true} props))))
+
+(defn qr-viewer-toolbar [name qr-value]
+  [react/view styles/account-toolbar
+   [react/view styles/toolbar-contents
+    [react/view styles/toolbar-action-container
+     [react/touchable-highlight {:on-press #(re-frame/dispatch [:navigate-back])}
+      (if platform/ios?
+        [react/view
+         [react/text {:style styles/toolbar-done-text-ios}
+         (i18n/label :t/done)]]
+        [react/view styles/toolbar-action-icon-container
+         [vi/icon :icons/close {:color :black}]])]]
+    [react/view styles/name-container
+     [react/text {:style           styles/name-text
+                  :number-of-lines 1} name]]
+    [react/view styles/toolbar-action-container
+     [react/touchable-highlight {:on-press #(list-selection/open-share {:message qr-value})}
+      [react/view styles/toolbar-action-icon-container
+       [vi/icon :icons/share {:color :black}]]]]]])
+
+(defn qr-viewer-body [qr-value dimensions]
+  [react/view {:style     styles/qr-code
+               :on-layout #(let [layout (.. % -nativeEvent -layout)]
+                             ;;TODO(goranjovic): check if we really really need to do this
```

Comment:
Could we address that before the merge?

Reply:
Apparently, we do need it. Removed the comment.